constellations/gemini-admin/src/user/userClass.js

```diff
@@ -0,0 +1,212 @@
+                var classes = [];
+                res.data.forEach((c) => {
+                    if (searchArray.has(c.classId)) {
+                        userClasses.push(c);
+                    } else {
+                        classes.push(c);
+                    }
+                });
+
+                this.setState({
+                    userClasses: userClasses,
+                    classes: classes,
+                });
+            })
+            .catch((err) =>
+                alert("Could not fetch classes: " + err.response.data)
+            );
+    };
+
+    onClassChange = (e) => {
+        this.setState({
+            classId: e.target.value,
+        });
+    };
+
+    onClickEnroll = () => {
+        const userClass = {
+            userId: parseInt(this.state.id),
+            classId: parseInt(this.state.classId),
+        };
+
+        API.post("api/userclasses/create", userClass)
+            .then(() => {
+                this.fetchUser();
+            })
+            .catch((err) =>
+                alert("Could not enroll in class: " + err.response.data)
+            );
+    };
+
+    render = () => {
+        const user = this.state.user;
+
+        var fullName = user.firstName;
+        if (user.middleName) {
+            fullName += " " + user.middleName + " " + user.lastName;
+        } else {
+            fullName += " " + user.lastName;
+        }
+
+        const rows = this.state.userClasses.map((c, index) => {
+            const enrollStatus = this.state.enrollStatuses[index];
+            const fullState = c.fullState;
+            let state = "";
+            let firstSpace = <button></button>;
+            let secondSpace = <button></button>;
+            switch (fullState) {
+                case 0:
+                    state = "Empty";
+                    firstSpace = <button>Reject</button>;
+                    secondSpace = <button>Accept</button>;
+                    break;
+                case 1:
+                    status = "Full";
+                    firstSpace = <button>Reject</button>;
+                    break;
+                case 2:
+                    state = "Almost Full";
+                    firstSpace = <button>Reject</button>;
+                    secondSpace = <button>Accept</button>;
+                    break;
+            }
+            if (enrollStatus == "Enrolled") {
+                firstSpace = <button></button>;
+                secondSpace = <button>Unenroll</button>;
+            }
+
+            return (
+                <div className="row" key={index}>
+                    <span className="large-column">{c.classId}</span>
+                    <span className="column">{enrollStatus}</span>
+                    <span className="column">{state}</span>
+                    <span className="button">{firstSpace}</span>
+                    <span className="button">{secondSpace}</span>
+                </div>
+            );
+        });
```

Comment:
For readability, maybe you can separate this to its own function or even own component.
We want to keep the render function small.